Source/Request.swift

```diff
@@ -86,6 +82,12 @@ open class Request {
         var redirectHandler: RedirectHandler?
         /// `CachedResponseHandler` provided to handle caching responses.
         var cachedResponseHandler: CachedResponseHandler?
+        /// Response serialization closures that handle parsing responses.
+        var responseSerializers: [() -> Void] = []
+        /// Current response serialization closure index used to track iteration.
+        var responseSerializerIndex: Int = 0
```

Comment:
I'm not huge fan of tracking an index like this, but Swift doesn't have any other suitable type. 😕

Reply:
Oh man me too! I was actually able to just get rid of it altogether in 2ec42dd2 by using the counts on the two arrays.